README.md

```diff
@@ -53,6 +53,33 @@ def params = [service: [code: "FFC", name: "Future Farming Services", type: "FFC
 ### destroyInfrastructure
 This will tear down infrastructure provisioned by `provisionInfrastructure` and should be called when a PR is closed. It only requires the repo name and pr number and a single call to tear down all the infrastructure created, which may have taken several calls to `provisionInfrastructure`
 
+### provisionPrRoleAndSchema
+
+Creates a Postgres database role and schema for use by a PR.
+
+It takes six parameters:
+- a string describing the Jenkins secret text credentials ID that contains database host to connect to
+- a string describing the database name that the schema will be created in
+- a string describing the Jenkins usernamePassword credentials ID that contains the database user and password
+- a string describing the Jenkins usernamePassword credentials ID that contains the password for the role that will be created
+- a string describing the PR number, e.g. `53`
+- a boolean to determine if the SQL commands to create the PR role and schema include the `IF NOT EXISTS` logic. If `true` the SQL commands **will not** error if the schema and/or role already exist. If `false` the SQL commands **will** error if the schema and/or role already exist and the pipeline wil exit.
```

Comment:
This flexibility seems to add to the complexity of the functionality, would it be possible to have a convention that `IF NOT EXISTS` be specified for these SQL commands?

Reply:
Agreed, it could be cleaner. Updated to implement the following:

1. `provision` by default doesn't apply the `IF NOT EXISTS` but you can override with an optional boolean parameter
2. `destroy` always applies IF EXISTS . There is no parameterisation, I couldn't think of a case where you want it to fail on dropping a non-existant role/schema